bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/internal/discovery/BroadlinkDiscoveryService.java

```diff
@@ -0,0 +1,105 @@
+
+    private final Logger logger = LoggerFactory.getLogger(BroadlinkDiscoveryService.class);
+    private int foundCount = 0;
+
+    public BroadlinkDiscoveryService() {
+        super(BroadlinkBindingConstants.SUPPORTED_THING_TYPES_UIDS_TO_NAME_MAP.keySet(), 10, true);
+        logger.info("BroadlinkDiscoveryService - Constructed");
+    }
+
+    public void startScan() {
+        foundCount = 0;
+        logger.warn("BroadlinkDiscoveryService - Beginning Broadlink device scan...");
+        DiscoveryProtocol.beginAsync(this, 10000L, this);
+    }
+
+    public void onDiscoveryFinished() {
+        logger.info("Discovery complete. Found {} Broadlink devices", foundCount);
+    }
+
+    protected synchronized void stopScan() {
+        super.stopScan();
+        removeOlderResults(getTimestampOfLastScan());
+    }
+
+    public void onDataReceived(String remoteAddress, int remotePort, String remoteMAC, ThingTypeUID thingTypeUID,
+            int model) {
+        logger.info("Data received during Broadlink device discovery: from {}:{} [{}]", remoteAddress, remotePort,
+                remoteMAC);
+        foundCount++;
+        discoveryResultSubmission(remoteAddress, remotePort, remoteMAC, thingTypeUID, model);
+    }
+
+    private void discoveryResultSubmission(String remoteAddress, int remotePort, String remoteMAC,
+            ThingTypeUID thingTypeUID, int model) {
+        String modelAsHexString = String.format("%x", model);
+        if (logger.isDebugEnabled()) {
+            logger.debug("Adding new Broadlink device ({} => {}) at {} with mac '{}' to Smarthome inbox",
+                    modelAsHexString, thingTypeUID, remoteAddress, remoteMAC);
+        }
+        Map<String, Object> properties = new HashMap<String, Object>(6);
+        properties.put("ipAddress", remoteAddress);
+        properties.put("port", Integer.valueOf(remotePort));
+        properties.put("mac", remoteMAC);
+        properties.put("deviceType", modelAsHexString);
+        ThingUID thingUID = new ThingUID(thingTypeUID, remoteMAC.replace(":", "-"));
+        if (logger.isDebugEnabled()) {
+            logger.debug("Device '{}' discovered at '{}'.", thingUID, remoteAddress);
+        }
+
+        if (BroadlinkBindingConstants.SUPPORTED_THING_TYPES_UIDS_TO_NAME_MAP.containsKey(thingTypeUID)) {
+            notifyThingDiscovered(thingTypeUID, thingUID, remoteAddress, properties);
+        } else {
+            logger.error("Discovered a {} but do not know how to support it at this time :-(", thingTypeUID);
+        }
+    }
+
+    private void notifyThingDiscovered(ThingTypeUID thingTypeUID, ThingUID thingUID, String remoteAddress,
+            Map<String, Object> properties) {
+        String deviceHumanName = BroadlinkBindingConstants.SUPPORTED_THING_TYPES_UIDS_TO_NAME_MAP.get(thingTypeUID);
+        String label = deviceHumanName + " [" + remoteAddress + "]";
+        DiscoveryResult result = DiscoveryResultBuilder.create(thingUID).withThingType(thingTypeUID)
```

Comment:
Can you add https://www.openhab.org/docs/developer/bindings/#representation-property